src/Sylius/Behat/Context/Domain/ProductContext.php

```diff
@@ -74,6 +105,14 @@ public function iShouldBeNotifiedThatThisVariantIsInUseAndCannotBeDeleted()
     }
 
     /**
+     * @When I should be notified that this product is in use and cannot be deleted
+     */
+    public function iShouldBeNotifiedThatThisProductIsInUseAndCannotBeDeleted()
+    {
+        expect($this->sharedStorage->get('last_exception'))->toHaveType(DBALException::class);
```

Comment:
Actually I would merge these steps into one. Because if a product is in use it mean that any of it's variants was purchased in an order. The message will be the same.